frontend/src/app/main-table/volumes-table/volume-table.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, OnInit, ViewChild } from "@angular/core";
+import { MatSort } from "@angular/material/sort";
+import { MatTableDataSource } from "@angular/material/table";
+import { MatDialog } from "@angular/material/dialog";
+import { Subscription } from "rxjs";
+import { first } from "rxjs/operators";
+import { isEqual } from "lodash";
+
+import { NamespaceService } from "src/app/services/namespace.service";
+import { KubernetesService } from "src/app/services/kubernetes.service";
+import { ExponentialBackoff } from "src/app/utils/polling";
+import { Volume } from "../../utils/types";
+import { ConfirmDialogComponent } from "../resource-table/confirm-dialog/confirm-dialog.component";
+
+interface Pvc {
+  pvc: Volume;
+  ismounted: boolean
+}
+
+@Component({
+  selector: "app-volume-table",
+  templateUrl: "./volume-table.component.html",
+  styleUrls: ["./volume-table.component.scss"]
+})
+
+export class VolumeTableComponent implements OnInit {
+  @ViewChild(MatSort) sort: MatSort;
+
+  // Logic data
+  resources = [];
+  currNamespace = "";
+  usedPVCs: Set<string> = new Set<string>();
+  showNameFilter = false;
+  pvcs: Volume[] = [];
+  customPvcs: Pvc[] = [];
+
+  subscriptions = new Subscription();
+  poller: ExponentialBackoff;
+
+  // Table data
+  displayedColumns: string[] = [
+    "name",
+    "namespace",
+    "isMounted",
+    "actions"
+  ];
+  dataSource = new MatTableDataSource();
+
+
+  constructor(
+    private namespaceService: NamespaceService,
+    private k8s: KubernetesService,
+    private dialog: MatDialog
+  ) { }
+
+  ngOnInit() {
+    this.dataSource.sort = this.sort;
+
+    this.poller = new ExponentialBackoff({ interval: 2000, retries: 3 });
+    const resourcesSub = this.poller.start().subscribe(() => {
+      if (!this.currNamespace) {
+        return;
+      } 
+
+      //TODO: make sure the getResource was ran first
+      this.k8s.getVolumes(this.currNamespace).subscribe(pvcs => {
+        if(isEqual(this.pvcs, pvcs)){
+          return;
+        }
+        this.pvcs = pvcs;
+        this.updatePvcStatus(true); //force the update of pvc status     
+      }); 
+      this.updatePvcStatus(false);
```

Comment:
The two requests are independent, so they can be done in parallel. If you decouple the concepts at play here, I think the implementation will become clearer:

* Requests, streams, etc. go fetch, or receive, updates in state – akin to what a controller handles in MVC
* State is passed to components for rendering as a (changing) data structure – akin to view in MVC
* View renders meaningful UI for the various possible states of its input data

So you can separate the concerns of setting the data structure on the component context from the rendering the current state of the context.